src/dos/drive_local.cpp

```diff
@@ -232,6 +232,79 @@ std::string localDrive::MapDosToHostFilename(const char* const dos_name)
 	return dirCache.GetExpandNameAndNormaliseCase(newname);
 }
 
+void localFile::ReplaceNativeFileHandle(const NativeFileHandle new_handle, const std_fs::path& new_path)
+{
+	close_native_file(file_handle);
+	file_handle = new_handle;
+	path = new_path;
+}
+
+[[maybe_unused]] static void copy_file_contents(const NativeFileHandle source_handle, const std_fs::path& dest_path)
+{
+	const auto dest_handle = open_native_file(dest_path, true);
+	if (dest_handle == InvalidNativeFileHandle) {
+		LOG_WARNING("FS: Failed to open file '%s'", dest_path.string().c_str());
+		return;
+	}
+	seek_native_file(source_handle, 0, NativeSeek::Set);
+	uint8_t buffer[4096] = {};
+	while (true) {
+		const auto read_result = read_native_file(source_handle, buffer, sizeof(buffer));
+		if (read_result.num_bytes < 1) {
+			break;
+		}
+		write_native_file(dest_handle, buffer, read_result.num_bytes);
+	}
+	close_native_file(dest_handle);
+}
+
+// Windows specific hack. This handles the case where a file gets unlinked with file handles still open.
+// FILE_SHARE_DELETE gets mostly the correct behavior here except it prevents a new file with the same name from being created.
+// See issue with the game Abuse here: https://github.com/dosbox-staging/dosbox-staging/issues/4123
+// POSIX gives us the behavior we want by default so nothing needs to be done on non-Windows OSs.
+#if defined(WIN32)
+static void move_open_handles_to_temp(const std_fs::path& path)
+{
+	std_fs::path temp_filename = {};
+	for (auto& file : Files) {
+		auto local_file = dynamic_cast<localFile*>(file.get());
+		if (!local_file || local_file->GetPath() != path) {
+			continue;
+		}
+		auto file_pos = get_native_file_position(local_file->file_handle);
+		if (file_pos == NativeSeekFailed) {
+			LOG_WARNING("FS: Failed to get current file pos. Setting to zero and hoping for the best :)");
```

Comment:
Let's keep the log messages a bit more formal 😄 "Setting to zero" will suffice, and it's "file position" 😏